### PR TITLE
Add a native GATK implementation for 2bit references, and remove the dependency on the ADAM library

### PR DESCRIPTION
**Pull request by droazen**

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/utils/ByteMapIntervalPileup.java [hunk 1]
```diff
@@ -344,10 +344,11 @@ private Element(final ByteMapIntervalPileup pileup, final GATKRead read, final i
             inserts = consolidateInserts(pileup.insertsBuffer, pileup.insertsBufferOffsets);
         }
 
+        @SuppressWarnings("unchecked")
         private static Int2ObjectMap<Insert> consolidateInserts(final List<Insert> inserts, final IntList offsets) {
             final int size = inserts.size();
             if (size == 0) {
-                return Int2ObjectMaps.emptyMap();
+                return (Int2ObjectMap<Insert>)Int2ObjectMaps.EMPTY_MAP;
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/ByteMapIntervalPileup.java:351]: Pin fastutil to `7.0.13` and get rid of this

File: src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java [hunk 2]
```diff
@@ -472,6 +472,43 @@ public static GenomeLocSortedSet sortAndMergeIntervals(final GenomeLocParser par
...
+     * the same genomic territory but with all overlapping (and, if specified, adjacent) intervals merged.
+     *
+     * WARNING: Since this method does not take a sequence dictionary, it uses lexicographical ordering for the contig
+     * names. If this is not desirable, use the overload that takes GenomeLocs instead.
+     *
+     * @param raw List of intervals to merge. Does not need to be sorted.
+     * @param rule Whether to merge adjacent intervals, or just overlapping intervals
+     * @return A new list of merged disjoint intervals, sorted by start position
+     */
+    public static List<SimpleInterval> sortAndMergeIntervals(final List<SimpleInterval> raw, final IntervalMergingRule rule) {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java:486]: Strongly consider adding a check enforcing a single-contig restriction

File: src/main/java/org/broadinstitute/hellbender/utils/reference/TwoBitReference.java [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+    // The complete in-memory contents of the 2bit file. Initialized as read-only in the constructor
+    private final ByteBuffer rawBytes;
+
+    // Byte order to use, as defined in the 2bit header
+    private ByteOrder byteOrder;
+
+    // The total number of sequences in the file
+    private int sequenceCount;
+
+    private LinkedHashMap<String, TwoBitSequenceRecord> sequenceRecords;
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/reference/TwoBitReference.java:68]: Add a comment that the order does in fact matter here

**Review by lbergelson — APPROVED** [review]
Looks 2bit to me 👍

**Pull request merged**